src/2048-expert/index.js

```diff
@@ -28,7 +28,9 @@ function generateNewTile() {
   } else {
     // if a cell is already has a number, then find a new tile
     checkLost();
-    generateNewTile();
+    if(!checkisFull()){
+      generateNewTile();
+    }
```

Comment:
Doesn't calling `checkLost()` implicitly do this check already?

Reply:
checkLost() check the block is full to make winning/ losing condition. Therefore if the checkLost() decides that the game is still playable, the game will continue and generateNewTile() will occur. However, when you call genertareNewTile() with no space on the board, an error occurs for creating a new tile that exceeds the space.

Reply:
I think there is an edge case where we'd only generate 1 tile that should be addressed. Either we spawn both tiles or no tiles imo.

Reply:
Depends what others think though imo, if we'd prefer to spawn as many as we can then that could be valid too.